### PR TITLE
Add i18next translations module

### PR DESCRIPTION
**Pull request by nastikue**

### CODE REVIEW

File: client/src/components/LanguageSelector.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+class LanguageSelector extends Component {
+  render() {
+    const i18n = this.props.i18n;
+    const language = i18n.language;
+
+    return language
+    ? (
+    <ul>
+      <li onClick={() => i18n.changeLanguage("en")} style={{ color: language === "en" ? "blue" : "inherit"}}>EN</li>
```
Comment (aviskarkc10) [inline client/src/components/LanguageSelector.js]: let's store the supported languages in an array of objects and then we can render the `<li>` in a loop. Doing so would be lot better as we add more languages

File: client/src/containers/App.js [hunk 2]
```diff
@@ -1,9 +1,14 @@
-import React from 'react'
+import React, { Suspense } from 'react'
+
 import PatientBoard from './PatientBoard'
+import Loader from '../components/Loader'
+import LanguageSelector from "../components/LanguageSelector";
 
 const App = () => (
-  <div>
+  <Suspense fallback={Loader}>
```
Comment (aviskarkc10) [inline client/src/containers/App.js]: why do we need `Suspense` here?

File: client/src/components/LanguageSelector.js (hunk 1 above)
Comment (nastikue) [reply client/src/components/LanguageSelector.js]: Supported languages list can be received form i18n.config.whitelist. LanguageSelector component was added only for reviewers to test languages work correctly and will be removed before PR merge. Language selector task should be done in separate task when design for language switcher is provided.

File: client/src/containers/App.js (hunk 2 above)
Comment (nastikue) [reply client/src/containers/App.js]: Good question! We need Suspense to get translations when they are ready https://react.i18next.com/latest/usetranslation-hook http://prntscr.com/rkm4t1 . I've rechecked react Suspense component : it is an experimental feature https://reactjs.org/docs/concurrent-mode-suspense.html , that is why I will replace it with custom Suspense component. Thank you for paying attention to this moment!

Comment (aviskarkc10) [reply client/src/containers/App.js]: yeah the docs says this:
```
The useTranslation hook will trigger a Suspense if not ready (eg. pending load of translation files). You can set useSuspense to false if prefer not using Suspense.
```

But we haven't used `useTranslation` hook anywhere have we?

So I don't think using `Suspense` is necessary here?

Correct me if I am wrong

Comment (nastikue) [reply client/src/containers/App.js]: As classes components are used in the project useTranslation hook won't be used, instead withTranslation HOC will be implemented in the components that need translation, this HOC needs Suspense as well https://react.i18next.com/latest/withtranslation-hoc.

Comment (aviskarkc10) [reply client/src/containers/App.js]: okay. Thanks for clarifying :+1:

File: client/src/components/LanguageSelector.js (hunk 1 above)
Comment (aviskarkc10) [reply client/src/components/LanguageSelector.js]: This file is going to be deleted right?

File: client/src/containers/App.js (hunk 2 above)
Comment (nastikue) [reply client/src/containers/App.js]: Custom TranslationsSuspense was added, please check

File: client/src/components/LanguageSelector.js (hunk 1 above)
Comment (nastikue) [reply client/src/components/LanguageSelector.js]: Correct. When PR is approved, I will remove redundant component

Comment (aviskarkc10) [reply client/src/components/LanguageSelector.js]: Okay. Let's remove this file and I will remove the `Do Not Merge` as well as approve the PR

Comment (nastikue) [reply client/src/components/LanguageSelector.js]: LanguageSelector removed

**Comment by nthopinion** [comment]
Hi @nastikue , @aviskarkc10 
I cloned https://github.com/nastikue/covid19/tree/feature/i18next-translations to run the site locally to review the translation changes.
I did not see a menu option to select a different language. 

image
Yet to merge, please respond.

**Comment by nastikue** [comment]
Hi @nthopinion , there is no option in menu design for language, that is why lang selection that was added for review purposes was removed form the PR for now.

**Pull request merged**